src/CoreWCF.Primitives/src/CoreWCF/Security/RequestSecurityTokenResponse.cs

```diff
@@ -0,0 +1,708 @@
+                    #pragma warning suppress 56503
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ItemNotAvailableInDeserializedRSTR, "EntropyToken")));
+                }
+                return this.entropyToken;
+            }
+        }
+
+        public SecurityToken RequestedSecurityToken
+        {
+            get 
+            {
+                if (this.isReceiver)
+                {
+                    // PreSharp Bug: Property get methods should not throw exceptions.
+                    #pragma warning suppress 56503
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ItemNotAvailableInDeserializedRSTR, "IssuedToken")));
+                }
+                return this.issuedToken;
+            }
+            set
+            {
+                if (this.isReadOnly)
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ObjectIsReadOnly)));
+                this.issuedToken = value;
+            }
+        }
+
+        public SecurityToken RequestedProofToken
+        {
+            get
+            {
+                if (this.isReceiver)
+                {
+                    // PreSharp Bug: Property get methods should not throw exceptions.
+                    #pragma warning suppress 56503
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ItemNotAvailableInDeserializedRSTR, "ProofToken")));
+                }
+                return this.proofToken;
+            }
+            set
+            {
+                if (this.isReadOnly)
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ObjectIsReadOnly)));
+                this.proofToken = value;
+            }
+        }
+
+        public XmlElement RequestSecurityTokenResponseXml
+        {
+            get
+            {
+                if (!this.isReceiver)
+                {
+                    // PreSharp Bug: Property get methods should not throw exceptions.
+                    #pragma warning suppress 56503
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ItemAvailableInDeserializedRSTROnly, "RequestSecurityTokenXml")));
+                }
+                return this.rstrXml;
+            }
+        }
+
+        internal object AppliesTo
+        {
+            get
+            {
+                if (this.isReceiver)
+                {
+                    // PreSharp Bug: Property get methods should not throw exceptions.
+                    #pragma warning suppress 56503
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ItemNotAvailableInDeserializedRST, "AppliesTo")));
```

Comment:
Multiple properties in this file have an exception which has a string parameter which uses the property name. Change to nameof where appropriate